### PR TITLE
Improve Primitive array comparison

### PR DESCRIPTION
**Pull request by AdamGS**
Some inspiration from arrow, some trying to do less work/being more correct about validity

### CODE REVIEW

File: vortex-array/src/array/primitive/compute/compare.rs [hunk 1]
```diff
@@ -14,37 +15,75 @@ impl CompareFn for PrimitiveArray {
...
 ) -> BooleanBuffer {
-    let matches = lhs.iter().zip(rhs.iter()).map(|(lhs, rhs)| f(lhs, rhs));
-    BooleanBuffer::from_iter(matches)
+    const BLOCK_SIZE: usize = u64::BITS as usize;
+
+    let len = lhs.len();
+    let reminder = len % BLOCK_SIZE;
+    let block_count = len / BLOCK_SIZE;
+
+    let mut buffer = MutableBuffer::new(ceil(len, BLOCK_SIZE) * 8);
```
Comment (robert3005) [inline vortex-array/src/array/primitive/compute/compare.rs:58]: aren’t brackets off here?

Comment (robert3005) [reply vortex-array/src/array/primitive/compute/compare.rs:58]: Ignore me, misread what this function does

**Review by robert3005 — APPROVED** [review]

File: vortex-array/src/array/primitive/compute/compare.rs [hunk 2]
```diff
@@ -14,37 +15,75 @@ impl CompareFn for PrimitiveArray {
...
+            let r = f(lhs[idx], rhs[idx]);
+            packed_block |= (r as u64) << bit_idx;
+        }
+
+        unsafe {
+            buffer.push_unchecked(packed_block);
+        }
+    }
+
+    BooleanBuffer::new(buffer.into(), 0, len)
```
Comment (a10y) [inline vortex-array/src/array/primitive/compute/compare.rs:86]: this logic looks like it's mainly the same as `MutableBuffer::collect_bool`. couldn't we just do

```
let buffer = MutableBuffer::collect_bool(|idx| f(lhs[idx], rhs[idx]));
```

Comment (robert3005) [reply vortex-array/src/array/primitive/compute/compare.rs:86]: We could use this logic but I think we will end up changing this function quite a bit so I think better for now

**Pull request merged**